crates/governance/src/storage/proposal.rs

```diff
@@ -589,18 +589,57 @@ pub mod testing {
     }
 
     prop_compose! {
-        /// Generate an arbitrary PGF target
-        pub fn arb_pgf_target()(
+        /// Generate an arbitrary PGF internal target
+        pub fn arb_pgf_internal_target()(
             target in arb_non_internal_address(),
             amount in arb_amount(),
-        ) -> PGFTarget {
-            PGFTarget::Internal(PGFInternalTarget {
+        ) -> PGFInternalTarget {
+            PGFInternalTarget {
                 target,
                 amount,
-            })
+            }
+        }
+    }
+
+    prop_compose! {
+        /// Generate an arbitrary port ID
+        pub fn arb_ibc_port_id()(id in "[a-zA-Z0-9_+.\\-\\[\\]#<>]{2,128}") -> PortId {
```

Comment:
I think ports should be alphanumeric. the wallet might not be checking this, so it's probably irrelevant

Reply:
Thanks for reviewing. Either way works for me since I don't know enough about IBC specs. That being said, the ICS-24 spec seems to allow port numbers like this. See https://github.com/cosmos/ibc/blob/main/spec/core/ics-024-host-requirements/README.md#paths-identifiers-separators . Also note that the regex pattern above was the most general pattern I could find for which `PortId::new`'s internal validation (based on ICS-24) returned `Result::Ok`.

Reply:
if it's been working against a validation predicate, then it's probably ok to keep it as is